### ISSUE TITLE
Move the ViRelAY package into a source subfolder

### ISSUE THREAD

**Issue by lecode-official** [virelay/virelay#46]
There are several problems with keeping the ViRelAy package directory in the top-level of the repository. This is a hotly debated topic, but the gist is that using a `source` directory is better because:

- The current directory is implicitly added to `sys.path` and since users will never have the same working directory as the developer, this can lead to import problems.
- The developer will be forced to install the package, which will make it easier to find broken dependencies before distribution. We already recommend this in the documentation, but this way developers actually have no other choice. Also, even when installing the package as editable, not having a `source` directory would still mean that the current parent directory of the package would be added to `sys.path`.
- It prevents developers from importing the package in `setup.py` which is bad practice as it would create a hen or egg situation where dependencies might be imported before they are installed (in fact, we may want to switch to TOML projects as well, but that is separate issue).

In summary, there are less changes for user mistakes and less chances for tools to mix up code with non-code. Also, having a `source` directory is almost an industry standard.

For more information, please refer to the following blog post: [Packaging a python library](https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure).
